### PR TITLE
Add go 1.17

### PR DESCRIPTION
**Pull request by JAicewizard**
This should mirror the bump to go 1.16, but for go 1.17

### CODE REVIEW

File: etc/config/go.amazon.properties [hunk 1]
```diff
@@ -115,6 +122,10 @@ compiler.arm_gl116.exe=/opt/compiler-explorer/golang-1.16/go/bin/go
 compiler.arm_gl116.semver=1.16
 compiler.arm_gl116.goarch=arm
 compiler.arm_gl116.name=ARM gc 1.16
+compiler.arm_gl117.exe=/opt/compiler-explorer/golang-1.17/go/bin/go
+compiler.arm_gl117.semver=1.16
```
Comment (DmitriyMV) [inline etc/config/go.amazon.properties:126]: Should this be 1.17?

File: etc/config/go.amazon.properties [hunk 2]
```diff
@@ -75,7 +75,10 @@ compiler.gl1150.semver=1.15
 compiler.gl1150.goroot=/opt/compiler-explorer/golang-1.15/go
 compiler.gl1160.exe=/opt/compiler-explorer/golang-1.16/go/bin/go
 compiler.gl1160.semver=1.16
-compiler.gl1160.goroot=/opt/compiler-explorer/golang-1.16/go
+compiler.gl1170.goroot=/opt/compiler-explorer/golang-1.16/go
```
Comment (DmitriyMV) [inline etc/config/go.amazon.properties:78]: This should be probably be:
```suggestion
compiler.gl1160.goroot=/opt/compiler-explorer/golang-1.16/go
```

File: etc/config/go.amazon.properties [hunk 3]
```diff
@@ -75,7 +75,10 @@ compiler.gl1150.semver=1.15
...
+compiler.gl1170.exe=/opt/compiler-explorer/golang-1.17/go/bin/go
+compiler.gl1170.semver=1.17
+compiler.gl1170.goroot=/opt/compiler-explorer/golang-1.16/go
```
Comment (DmitriyMV) [inline etc/config/go.amazon.properties:81]: This should be probably be:

```suggestion
compiler.gl1170.goroot=/opt/compiler-explorer/golang-1.17/go
```

**Review by DmitriyMV — CHANGES_REQUESTED** [review]
I'm not a maintainer, but I noticed some errors.

**Comment by JAicewizard** [comment]
Yes thank you for checking! I should have looked a bit closer. Its so hard to see these errors with so much samy information.

It would be really nice if these ware generated, but the rate of change on these files is probably too low.

**Comment by RubenRBS** [comment]
Hi! Thank you so much for taking the time to keep Go up-to-date! (And thanks @DmitriyMV for the double-check!)

This is missing a PR in the sister infra repo. See how the 1.16 update did it here https://github.com/compiler-explorer/infra/pull/502

**Comment by JAicewizard** [comment]
done as well https://github.com/compiler-explorer/infra/pull/586

**Comment by RubenRBS** [comment]
Looks great. Again, thank you so much! This should be live Soon:tm:

**Pull request merged**